src/SocrataSelectQuery.php

```diff
@@ -47,115 +48,51 @@ public function setEndpoint(Endpoint $endpoint = NULL) {
   /**
    * Execute the query. In our case, this is actually the cURL request.
    *
-   * @param $type string
+   * @param string $type
    *   cURL request type, if not the default. The only acceptable value is
    *   currently 'metadata', but we might want others in the future?
    *
    * @return array
+   *   An array containing headers and response body
    */
   public function execute($type = NULL) {
     $retval = FALSE;
-
-    // @todo Change to use Guzzle client a la http://www.lohmeyer.rocks/blog/2015/10/20/0070-how-make-custom-guzzle-requests-drupal-8-modules.
-    // Create a new cURL resource.
-    $ch = curl_init();
-    if ($ch) {
-      // Pull and set custom curl options.
-      $curlopts = \Drupal::config('socrata.settings')->get('socrata_curl_options');
-      \Drupal::moduleHandler()->alter('socrata_curl_options', $curlopts);
-      curl_setopt_array($ch, $curlopts);
-
-      // Set required curl options.
-      $req_curlopts = array(
-        CURLOPT_HEADER => TRUE,
-        CURLOPT_RETURNTRANSFER => TRUE,
-      );
-      curl_setopt_array($ch, $req_curlopts);
-
-      // Execute the request and cleanup.
-      curl_setopt_array($ch, $curlopts);
-      do {
-        $retval = FALSE;
-
-        // Construct the URL, based on the $type parameter.
-        if ($type == 'metadata') {
-          // Note that 'metadata' here represents an entirely separate API and
-          // endpoint, and is different from the special fields of type
-          // 'meta_data' (mentioned below) that are returned from the normal
-          // dataset API/endpoint, but when using an old version of the API.
-          // @todo Add version number for "old".
-          $curl_url = $this->endpoint->getMetaDataURL();
-        }
-        else {
-          // Can build the SODA URL now that all the parameters have been set.
-          $curl_url = $this->endpoint->getSodaURL($this->params);
+    $client = \Drupal::httpClient();
+    try {
+      $options = \Drupal::config('socrata.settings')->get('socrata_curl_options');
+      \Drupal::moduleHandler()->alter('socrata_curl_options', $options);
+      if ($type == 'metadata') {
+        // Note that 'metadata' here represents an entirely separate API and
+        // endpoint, and is different from the special fields of type
+        // 'meta_data' (mentioned below) that are returned from the normal
+        // dataset API/endpoint, but when using an old version of the API.
+        // @todo Add version number for "old".
+        $url = $this->endpoint->getMetaDataURL();
+      }
+      else {
+        // Can build the SODA URL now that all the parameters have been set.
+        $url = $this->endpoint->getSodaURL($this->params);
+      }
+      _socrata_dbg($url);
+
+      $res = $client->get($url, [
+        'curl' => $options,
```

Comment:
Does this work?  Did you test adding options?

And even if it does, is it useful for the user to specify options for the Guzzle client?